### DISCUSSION TITLE
[FEATURE] Can the ReadWriteMany mode work without using NFS?

### DISCUSSION THREAD

**Discussion by w7team** [longhorn/longhorn#9924] [General]
由于NFS没有对元数据进行缓存，导致在处理大量小文件时非常影响性能。咱们目前已经实现了多个副本的一致性同步，能否在使用ReadWriteMany模式时，每个pod挂载时使用各自节点上的副本，给一种不采用NFS的方式。

Due to NFS not caching metadata, it greatly affects performance when dealing with a large number of small files. We have already implemented consistent synchronization of multiple replicas. Is it possible that when using the ReadWriteMany mode, each pod mounts its own replica on the respective node, providing a method that does not rely on NFS?

**Comment by ChanYiLin** [comment]
Unfortunately no, due to the limitation of the the current design of Longhorn.
One volume can only have one engine. The engine creates the device on the node and K8s mount the device into the pod.
In your proposal, that would be one pod with one engine and connect to one replica of the same volume.
But Longhorn maintains the consistency by dispatching the IO from engine to all replicas at the same time. Not by syncing the replicas afterward.
That is, in your proposal, all the replicas won't have data consistency.
